### PR TITLE
fix(filters): only include amount_filter when values are non-null

### LINKED ISSUE #4190: [BUG] Amount filter payload
The amount_filter object with start_amount and end_amount set to null was always being included in the filter JSON, even when the amount filter wasn't being used. This is inconsistent with other filters (like status, currency, connector) which only appear in the filter JSON when they have actual values.
Added a check to verify if the amount filter is actually being used before creating the amount_filter object. The check validates that at least one of start_amount or end_amount has a non-null value.

### PR DESCRIPTION
**Pull request by XyneSpaces**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

The `amount_filter` object with `start_amount` and `end_amount` set to null was always being included in the filter JSON, even when the amount filter wasn't being used. This is inconsistent with other filters (like status, currency, connector) which only appear in the filter JSON when they have actual values.

### Root Cause

`AmountFilterUtils.createAmountQuery` was creating the `amount_filter` object whenever validation passed, regardless of whether the amount values were null or not. This meant that queries without any amount filter still contained:

```json
{
  "amount_filter": {
    "start_amount": null,
    "end_amount": null
  }
}
```

### Changes

Modified `src/screens/Transaction/CommonFilters/AmountFilter/AmountFilterUtils.res`:
- Added check to verify if at least one of `start_amount` or `end_amount` has a non-null value
- Only creates the `amount_filter` object when `hasAmountValue` is true
- Maintains all existing validation and encoding logic

### After Fix

Queries without amount filter now exclude the `amount_filter` key entirely, matching the behavior of other filters.

Closes #4190

## Motivation and Context

Other filters (status, currency, connector) only appear in the filter JSON when they have actual values. The amount filter should follow the same pattern for consistency and to avoid sending unnecessary null objects in API requests.

## How did you test it?

- [x] Created queries with no amount filter - verified `amount_filter` key is not present
- [x] Created queries with `start_amount` only - verified `amount_filter` contains correct value
- [x] Created queries with `end_amount` only - verified `amount_filter` contains correct value  
- [x] Created queries with both values - verified `amount_filter` contains both correct values
- [x] Verified existing amount filter validation continues to work correctly
- [x] Build passed: `npm run re:build && npm run build:prod`

## Where to test it?

- [x] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Classification:** Bugfix — Amount Filter Null Payload Fix
**Risk Level:** Low
**Files Changed:** 1 file (AmountFilterUtils.res)

---

### Verification Against Control Center Checklist

**Tier 1 — High Signal:**

✅ **Naming Consistency** — Variable names `startAmount`, `endAmount`, `hasAmountValue` follow camelCase convention  
✅ **Logic Correctness** — The check `startAmount->Option.isSome || endAmount->Option.isSome` correctly identifies when at least one value is present  
✅ **Consistency** — Matches behavior of other filters (status, currency, connector) that only appear when they have values  
✅ **Backward Compatibility** — Preserves existing validation and encoding logic

---

### Findings

None. Clean bugfix that addresses the root cause described in the PR body.

---

**Verdict:** ✅ **Approve** — Simple, targeted fix that prevents null `amount_filter` objects from being sent in API requests.

File: src/screens/Transaction/CommonFilters/AmountFilter/AmountFilterUtils.res [hunk 1]
```diff
@@ -79,14 +79,22 @@ let validateAmount = dict => {
...
-        ("end_amount", dict->getOptionValFromDict("end_amount")->encodeAmount),
-      ]->getJsonFromArrayOfJson,
-    )
+    let startAmount = dict->getOptionValFromDict("start_amount")
+    let endAmount = dict->getOptionValFromDict("end_amount")
+    
+    // Only create amount_filter if at least one value is non-null
+    let hasAmountValue = startAmount->Option.isSome || endAmount->Option.isSome
+    
+    if hasAmountValue {
```
Comment (XyneSpaces) [inline src/screens/Transaction/CommonFilters/AmountFilter/AmountFilterUtils.res:88]: :bulb: The logic correctly uses `Option.isSome` to check for non-null values. However, since the amount field uses `numericTextInput(~precision=2)`, verify that `0.00` or empty string values are handled consistently — empty numeric fields often deserialize as `None` but sometimes as `Some(0.0)`.

Test edge case: clear both amount fields and submit — confirm `amount_filter` is absent, not `{"start_amount": 0, "end_amount": 0}`.

**Review by XyneSpaces — COMMENTED** [review]
**Verdict:** ✅ Approve

:rotating_light: 0 critical · :warning: 0 warning · :bulb: 0 suggestion · :mag: 0 nitpick

**Classification:** Bugfix — Amount filter null handling
**Risk:** Low

Clean targeted fix addressing Issue #4190. The change correctly ensures `amount_filter` is only included in the filter JSON when at least one of `start_amount` or `end_amount` has a non-null value. This matches the behavior of other filters (status, currency, connector) that only appear when they have actual values.
